src/main/java/com/benmorant/game/holycraap/model/entity/Item.java

```diff
@@ -25,4 +26,90 @@
   private int slot;
 
   private double price;
+
+  @ManyToOne(fetch = FetchType.LAZY)
+  @JsonBackReference
+  private People people;
+
+  public Item() {}
+
+  /** Constructeur surchargé pour Item (moins l'ID, auto-généré. */
+  public Item(String name, int slot, double price, People people) {
+    this.name = name;
+    this.slot = slot;
+    this.price = price;
+    this.people = people;
+  }
```

Comment:
I think I see what you're trying to do. I would advise against creating such constructors as they are not as extensible. As the number of arguments of `Item` grows, so will the number of arguments. You should the argument-less constructor and use the setter methods instead. Like this, you'll be able to use Lombok again.

Reply:
OK, NoArgsConstructor .
But I don't think I'll be able to use Lombok (not his @Data annotation anyway) The main problem was with the way Lombok generates equals, hashcode and toString methods... 
Maybe by just using @Getter, @Setter and @NoArgsConstructors (and @RequiredArgs for People) it would be fine...

=> DOING

I did a lot of research about that during the last couple of days + I studied how to build a docker config when your project is many repos (well, I tried to study, anyway), and I went through this : https://github.com/wkrzywiec/kanban-board/tree/master/kanban-app/src/main/java/com/wkrzywiec/medium/kanban/model and that's where I saw the distinction between entities and DTO





#3615MyLife